deepset_cloud_sdk/service/files_service.py

```diff
@@ -74,23 +107,77 @@ async def upload_file_paths(
-            ingested_files = 0
-            while ingested_files < len(file_paths):
-                if time.time() - start > timeout_s:
-                    raise TimeoutError("Ingestion timed out.")
-
-                upload_session_status: UploadSessionStatus = await self._upload_sessions.status(
-                    workspace_name=workspace_name, session_id=upload_session.session_id
-                )
-                ingested_files = (
-                    upload_session_status.ingestion_status.finished_files
-                    + upload_session_status.ingestion_status.failed_files
-                )
-                logger.info(
-                    "Waiting for ingestion to finish.",
-                    finished_files=upload_session_status.ingestion_status.finished_files,
-                    failed_files=upload_session_status.ingestion_status.failed_files,
-                    total_files=len(file_paths),
-                )
-                time.sleep(1)
+            await self._wait_for_finished(
+                workspace_name=workspace_name,
+                session_id=upload_session.session_id,
+                total_files=len(file_paths),
+                timeout_s=timeout_s,
+            )
+
+    async def upload_folder(
+        self, workspace_name: str, folder_path: Path, blocking: bool = True, timeout_s: int = 300
+    ) -> None:
+        """Upload a folder to a workspace.
+
+        Upload a folder via upload sessions to a selected workspace. If blocking is True, the function waits until
+        all files are uploaded and listed by deepsetCloud. If blocking is False, the function returns immediately after
+        the upload of the files is done. Note: It can take a while until the files are listed in deepsetCloud.
+
+        :param workspace_name: Name of the workspace to upload the files to.
+        :folder_path: Path to the folder to upload.
+        :blocking: If True, blocks until the ingestion is finished.
+        :timeout_s: Timeout in seconds for the blocking ingestion.
+        :raises TimeoutError: If blocking is True and the ingestion takes longer than timeout_s.
+        """
+        all_files = [path for path in folder_path.glob("**/*")]
+
+        file_paths = [
+            path
+            for path in all_files
+            if path.is_file() and ((path.suffix in [".txt", ".pdf"]) or path.name.endswith("meta.json"))
+        ]
+        if len(file_paths) < len(all_files):
+            logger.warning(
+                "Skipping files with unsupported file format.",
+                folder_path=folder_path,
+                skipped_files=len(all_files) - len(file_paths),
+            )
+
+        await self.upload_file_paths(
+            workspace_name=workspace_name, file_paths=file_paths, blocking=blocking, timeout_s=timeout_s
+        )
+
+    async def upload_texts(
+        self, workspace_name: str, dc_files: List[DeepsetCloudFiles], blocking: bool = True, timeout_s: int = 300
```

Comment:
I am not sure about the naming here and if we should call this dc_files. 
Pro: 
- files and meta are in on object.
- this single object (in a list) is typed 
Con:
- complexity to import these SDK specific data classes 
- passing multiple lists of text, meta and names requires Len checks and most of the time users might have dicts with text

Reply:
WDYT @rjanjua ?

Reply:
> I don't think the complexity is too much to import a datatype. Could we offer a helper method that would convert a set of files to a DeepsetCloufFile

I think we could also add validation the the data classes, default values etc too :)